packages/nextjs/components/scaffold-eth/Contract/MethodSelector.tsx

```diff
@@ -0,0 +1,124 @@
+import { useState } from "react";
+import { Abi } from "viem";
+import { ChevronDownIcon, ChevronRightIcon, XMarkIcon } from "@heroicons/react/24/outline";
+
+type AbiFunction = Extract<Abi[number], { type: "function" }>;
+
+type AbiFallback = Extract<Abi[number], { type: "fallback" }>;
+
+type SelectableAbiType = AbiFunction | AbiFallback;
+
+interface MethodSelectorProps {
+  methodsWithInputs: SelectableAbiType[];
+  abi: Abi;
+  onMethodSelect: (selectedMethods: string) => void;
+  removeMethod: (methodName: string) => void;
+}
+
+export const MethodSelector = ({ methodsWithInputs, abi, onMethodSelect, removeMethod }: MethodSelectorProps) => {
+  const [isReadCollapsed, setIsReadCollapsed] = useState(false);
+  const [isWriteCollapsed, setIsWriteCollapsed] = useState(false);
+
+  const readMethods = methodsWithInputs.filter(
+    (method): method is AbiFunction => method.type === "function" && method.stateMutability === "view",
+  );
+
+  const writeMethods = methodsWithInputs.filter(
+    (method): method is AbiFunction => method.type === "function" && method.stateMutability !== "view",
+  );
```

Comment:
Similar to https://github.com/BuidlGuidl/abi.ninja/pull/36/files#r1466817144 just to be sure we can do this : 

```ts
method.stateMutability !== "view" && method.stateMutability !== "pure"
```